crates/volta-core/src/run/binary.rs

```diff
@@ -167,11 +167,15 @@ impl DefaultBinary {
             None => return Ok(None),
         };
 
-        if bin_config_file.exists() {
-            let bin_config = BinConfig::from_file(bin_config_file)?;
-            DefaultBinary::from_config(bin_config, session).map(Some)
-        } else {
-            Ok(None) // no config means the tool is not installed
+        match BinConfig::from_file(bin_config_file) {
+            Err(error) => {
+                if error.is_not_found_error_kind() {
+                    Ok(None)
+                } else {
+                    Err(error)
+                }
+            }
+            Ok(bin_config) => DefaultBinary::from_config(bin_config, session).map(Some),
```

Comment:
Same here, this could be:

```rust
match BinConfig::from_file_if_exists(bin_config_file)? {
    Some(bin_config) => DefaultBinary::from_config(bin_config, session).map(Some),
    None => Ok(None),
}
```

Reply:
Done